src/test/java/uk/gov/justice/digital/delius/service/ConvictionServiceTest.java

```diff
@@ -186,6 +191,96 @@ public void addingACourtCaseCreatesAConvictionEventWithNewIndexMatchingNumberOfC
         verify(spgNotificationService).notifyNewCourtCaseCreated(any(Event.class));
     }
 
+    @Nested
+    class GetConvictionsWithRar {
+
+        @Test
+        public void convictionsReturnedIgnoresSoftDeletedEvents() {
+            when(eventRepository.findByOffenderIdAndActiveTrue(1L))
+                .thenReturn(asList(
+                    aEvent().toBuilder().eventId(99L).offenderId(1L).softDeleted(1L).build()
+                ));
+
+            assertThat(convictionService.convictionsWithActiveRequirementFor(1L, "F"))
+                .isEmpty();
+        }
+
+        @Test
+        public void convictionsReturnedFiltersEventsWithNoDisposals() {
+            when(eventRepository.findByOffenderIdAndActiveTrue(1L))
+                .thenReturn(asList(
+                    aEvent().toBuilder().eventId(99L).offenderId(1L).softDeleted(0L).build()
+                ));
+
+            assertThat(convictionService.convictionsWithActiveRequirementFor(1L, "F"))
+                .isEmpty();
+        }
+
+        @Test
+        public void convictionsReturnedFiltersEventsWithNoRequirements() {
+            when(eventRepository.findByOffenderIdAndActiveTrue(1L))
+                .thenReturn(asList(
+                    aEvent().toBuilder().eventId(99L).offenderId(1L).softDeleted(0L)
+                        .disposal(Disposal.builder().disposalId(98L).requirements(emptyList()).build())
+                        .build()
+                ));
+
+            assertThat(convictionService.convictionsWithActiveRequirementFor(1L, "F"))
+                .isEmpty();
+        }
+
+        @Test
+        public void convictionsReturnedFiltersEventsWithNoActiveRequirements() {
+            when(eventRepository.findByOffenderIdAndActiveTrue(1L))
+                .thenReturn(asList(
+                    aEvent().toBuilder().eventId(99L).offenderId(1L).softDeleted(0L)
+                        .disposal(Disposal.builder().disposalId(98L).requirements(
+                            singletonList(Requirement.builder().activeFlag(0L).build())).build())
+                        .build()
+                ));
+
+            assertThat(convictionService.convictionsWithActiveRequirementFor(1L, "F"))
+                .isEmpty();
+        }
+
+        @Test
+        public void convictionsReturnedFiltersEventsWithNoRarRequirements() {
+            when(eventRepository.findByOffenderIdAndActiveTrue(1L))
+                .thenReturn(asList(
+                    aEvent().toBuilder().eventId(99L).offenderId(1L).softDeleted(0L)
+                        .disposal(Disposal.builder().disposalId(98L).requirements(
+                            singletonList(Requirement.builder().activeFlag(1L).requirementTypeMainCategory(
+                                RequirementTypeMainCategory.builder().code("X").build()).build())).build())
+                        .build()
```

Comment:
Is it worth pulling a named code out of the RequirementTypeMainCategory entity?

Reply:
Will create as a constant in the test class as it's only needed in the test class as a non RAR example.